From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: deepak1556 <hop2deep@gmail.com>
Date: Sun, 22 Nov 2020 14:07:01 -0800
Subject: Make autocorrect=off and spellcheck=false disable Touch Bar typing
 suggestions.

https://chromium-review.googlesource.com/c/chromium/src/+/1917603

Patch cannot be upstreamed refer to discussion in https://issues.chromium.org/issues/40719955,
however the patch can be removed once mac with touch bar support reaches EOL.

diff --git a/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm b/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm
index 3d5cf1f31281e8412e6cdb441e6aa7954ad2d601..c67e3b774a27bd01026ef318629fa15e877a3616 100644
--- a/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm
+++ b/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm
@@ -2670,7 +2670,9 @@ - (void)invalidateTouchBar {
 }
 
 - (NSTouchBar*)makeTouchBar {
-  if (_textInputType != ui::TEXT_INPUT_TYPE_NONE) {
+  if (_textInputType != ui::TEXT_INPUT_TYPE_NONE &&
+      !(_textInputFlags & blink::kWebTextInputFlagAutocorrectOff) &&
+      !(_textInputFlags & blink::kWebTextInputFlagSpellcheckOff)) {
     _candidateListTouchBarItem = [[NSCandidateListTouchBarItem alloc]
         initWithIdentifier:NSTouchBarItemIdentifierCandidateList];
 
